tonic/src/status.rs

```diff
@@ -580,6 +580,7 @@ impl Status {
             .headers_mut()
             .insert(http::header::CONTENT_TYPE, GRPC_CONTENT_TYPE);
         self.add_header(response.headers_mut()).unwrap();
+        response.extensions_mut().insert(self);
```

Comment:
Am I missing something (its been a while since I looked at this code) but can you not extract the information directly from the headers?

Reply:
You can pass information to the middleware via headers with additional effort when the status is created. Like I said in the PR description, passing more complex types always involves a serialization step (since headers are just bytes) and a deserialization step if a middleware accesses this information.